app/src/main/java/com/example/tasky/feature_authentication/data/validation/UserDataValidatorImpl.kt

```diff
@@ -0,0 +1,62 @@
+package com.example.tasky.feature_authentication.data.validation
+
+import android.util.Patterns
+import com.example.tasky.feature_authentication.domain.validation.UserDataValidator
+
+class UserDataValidatorImpl : UserDataValidator {
+
+    private var errorMsg: String = ""
+    override fun validateFullName(fullName: String): Boolean {
+        if(fullName.isBlank()) {
```

Comment:
Validating user data is rather business logic than data related. Therefore, this would rather belong in the domain layer. The only thing that makes sense to abstract is the email validation, since that requires an Android import.

Something like this:
*In domain*
```
interface EmailPatternValidator {
    fun isValidEmailPattern(email: String): Boolean
}
```

*In data*
```
class EmailPatternValidatorImpl: EmailPatternValidator {

    override fun isValidEmailPattern(email: String): Boolean {
        return Patterns.EMAIL_ADDRESS.matcher(email).matches()
    }
}
```

Then inject an `EmailPatternValidator` into your domain level validation class and use there.